From d785201ef3b0ee0d826c6a95130617d6798d71d9 Mon Sep 17 00:00:00 2001
From: Mitchell Orsucci <mitchellorsucci@gmail.com>
Date: Mon, 23 Oct 2017 08:28:03 -0700
Subject: [PATCH] Fix shared libraries loading during backup

When backup or restore is triggered on a non running
application the shared libraries are not loaded. This
will result in ClassNotFoundException when launching
an application referenced a shared library. Fix is to
add a GET_SHARED_LIBRARIES flag when calling
getApplicationInfo, then loading of shared library
when application launch is triggered by backup.

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 70c2e61e57ae..577cdbce339c 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -18559,7 +18559,7 @@ public boolean bindBackupAgent(String packageName, int backupMode, int userId) {
         IPackageManager pm = AppGlobals.getPackageManager();
         ApplicationInfo app = null;
         try {
-            app = pm.getApplicationInfo(packageName, 0, userId);
+            app = pm.getApplicationInfo(packageName, STOCK_PM_FLAGS, userId);
         } catch (RemoteException e) {
             // can't happen; package manager is process-local
         }
